### DISCUSSION TITLE
Slow debugger startup? (pycharm windows)

### DISCUSSION THREAD

**Discussion by kfalcami** [zauberzeug/nicegui#3921] [Q&A]
### Question

I'm wondering if anybody else has noticed this: even a simple nicegui app takes a long time to start in the pycharm debugger, e.g. 48 secs vs. the same app starting in 13 secs without the debugger.  I'm a long-time pycharm user before I started using nicegui and this is a much bigger difference than I've seen in my other python development.

**Comment by falkoschindler** [comment]
Hi @kfalcami, we heard about similar problems with PyCharm, e.g. #3344. Even 13 seconds is pretty slow.

How does a plain FastAPI app behave?

server.py:
```py
import uvicorn
from fastapi import FastAPI

app = FastAPI()

@app.get("/")
async def root():
    return {"message": "Hello World"}

if __name__ == "__main__":
    uvicorn.run("server:app", host="0.0.0.0", port=8000, reload=True)
```

**Reply by kfalcami** [reply]
For your server.py time to "Application startup complete.": without debugger 3 secs, with debugger 38 secs

I also tried adding the frontend.py code (and call to init()) from the fastapi example in the nicegui repo: without debugger 6 secs, with debugger 63 secs

(and sorry the 13 secs was due to some init code I forgot to remove from my simple app, now it's also down to 3 secs without debugger)

**Comment by reinderien** [comment]
Even a do-nothing `print()` script takes 3-5 seconds for the debugger to start up for me. I don't see why this should be the case. The debugger shows command line

```none
.venv12\Scripts\python.exe -X pycache_prefix=C:\Users\me\AppData\Local\JetBrains\PyCharmCE2024.3\cpython-cache "C:/Program Files/JetBrains/PyCharm Community Edition 2021.2/plugins/python-ce/helpers/pydev/pydevd.py" --multiprocess --qt-support=auto --port 29781 --file scratch.py 
```

Surely we can do better. But of course this doesn't have anything to do with `nicegui`.
